### PR TITLE
Refactor assembly.status module

### PR DESCRIPTION
**Pull request by JAlvarezJarreta**
- moved part of the functionality to the argument parser to reduce code complexity
- added some additional type hints
- adjusted documentation

Still a draft until unit tests are provided, but would like to review the code with @ens-LCampbell first.

### CODE REVIEW

**Review by ens-LCampbell — APPROVED** [review]
Nice refactor Jorge thank you. I learn better practices as I review your code 👍 No issues from my perspective looks good to me

**Comment by JAlvarezJarreta** [comment]
Thank you @ens-LCampbell 😊 To be completely honest, not sure if what I'm doing with the parsers is the standard or a good practice, but it felt more modular.

File: src/python/tests/assembly/test_status.py [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+        db_factory(data_dir / "another_core_db", "another_core_db"),
+        db_factory(data_dir / "yet_another_core_db", "yet_another_core_db"),
+    ]
+    tmp_file = tmp_path / "core_dbs_list.txt"
+    with tmp_file.open("w") as fin:
+        for db in test_dbs:
+            fin.write(f"{db.dbc.db_name}\n")
+    test_server = make_url(request.config.getoption("server"))
+    accessions = fetch_accessions_from_core_dbs(tmp_file, test_server)
+    assert not DeepDiff(accessions, {f"{os.environ['USER']}_my_core_db.db": "GCF_001194135.2"})
```
Comment (ens-LCampbell) [inline src/python/tests/assembly/test_status.py:256]: Part of this issue appears to be related to SQLAlchemy 1.4.45 vs 2.0. I was able to get all tests to pass with --server, but had to alter the deepdiff line:256....  {f"{os.environ['USER']}_my_core_db.db -> {}_my_core_db. My venv has alchemy v1.4.45 installed.

File: src/python/ensembl/io/genomio/assembly/status.py [hunk 2]
```diff
@@ -117,188 +134,155 @@ def singularity_image_setter(sif_cache_dir: Path, datasets_version: str) -> Clie
...
+        server_url: Database server URL.
 
     Returns:
-        Dict of core name(s) (key) and its INSDC assembly.accession (value)
+        Dict of core database names (key) and their corresponding INSDC assembly accession (value).
     """
 
     core_accn_meta = {}
-    core_list_count = len(database_names)
+    num_lines = 0
```
Comment (ens-LCampbell) [inline src/python/ensembl/io/genomio/assembly/status.py:171]: not a fan of num_lines var declaration. Can we use 'database_count' ?

File: src/python/ensembl/io/genomio/assembly/status.py [hunk 3]
```diff
@@ -117,188 +134,155 @@ def singularity_image_setter(sif_cache_dir: Path, datasets_version: str) -> Clie
...
-            count_accn_found += 1
-            asm_accession = query_result.pop()[0]
-            logging.info(f"{core} -> assembly.accession[{asm_accession}]")
-            core_accn_meta[core] = asm_accession
-        else:
-            logging.warning(f"Core {core} has {len(query_result)} assembly.accessions")
+    with Path(src_file).open("r") as fin:
+        for line in fin.readlines():
+            core_db = line.strip()
+            num_lines += 1
```
Comment (ens-LCampbell) [inline src/python/ensembl/io/genomio/assembly/status.py:177]: same as above "database_count"

File: src/python/ensembl/io/genomio/assembly/status.py [hunk 4]
```diff
@@ -117,188 +134,155 @@ def singularity_image_setter(sif_cache_dir: Path, datasets_version: str) -> Clie
...
+            elif len(query_result) == 1:
+                count_accn_found += 1
+                asm_accession = query_result.pop()[0]
+                logging.info(f"{core_db} -> assembly.accession[{asm_accession}]")
+                core_accn_meta[core_db] = asm_accession
+            else:
+                logging.warning(f"Core {core_db} has {len(query_result)} assembly.accessions")
 
-    logging.info(f"From initial input cores ({core_list_count}), obtained ({count_accn_found}) accessions")
+    logging.info(f"From initial input core databases ({num_lines}), obtained ({count_accn_found}) accessions")
```
Comment (ens-LCampbell) [inline src/python/ensembl/io/genomio/assembly/status.py:195]: "database_count"

**Review by ens-LCampbell — APPROVED** [review]
Ran the tests a few times and had a good look over them, looked a-ok to me. All tests passed (including the one that was failing) once I removed the '.db. suffix.

**Pull request merged**